### ISSUE TITLE
Learning PPO2 for a custom environment takes a long time 

### ISSUE THREAD

**Issue by noyan72** [hill-a/stable-baselines#783]
I use PPO2 for the custom environment. But with more time step (5M), there is no convergence toward positive rewards.
I use Tensorboard to see the reward in each episode. Also, subprocvecenv for multiprocessing with num_cpu=4. Each episode with n_step=128  takes about 180 seconds.
How to get the optimal value of parameters in PPO2? 
How to speed up learning؟

System Info
Library installed using pip
Python version 3.6.9
Tensorflow version 1.14.0

**Comment by araffin** [comment]
Hello,

I recommend you to read the [tips and tricks](https://stable-baselines.readthedocs.io/en/master/guide/rl_tips.html) in the doc and take at look at the rl zoo. In short, you should take a look at hyperparameters for similar problem, and do hyperparameter optimization (and maybe choose another algorithm). 

Note that as mentioned, we don't do any technical support in the issues. Stack overflow or reddit is a better place for that ;)

**Comment by noyan72** [comment]
OK. I read the reinforcement learning tips and tricks again. For
normalization should the reward and observation had zero mean and one
variance? in my work the reward is between 1 and -1. I can not change
algorithm, because my action space is continuous and as we know ppo2 has
better performance than TRPO or A2C or GA3C, A3C algorithms.  

I have been reading your library for about 5 months and I am trying to
solve my problem with it. Recently, a new library called mushroom-rl has
been introduced. But I couldn't take the risk and use it. I will do my
best to answer with your library. 

Thanks alot for helping 

On 2020-04-05 15:16, Antonin RAFFIN wrote:

> Closed #783 [1]. 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [2], or unsubscribe [3].
 

Links:
------
[1] https://github.com/hill-a/stable-baselines/issues/783
[2]
https://github.com/hill-a/stable-baselines/issues/783#event-3202118952
[3]

**Comment by araffin** [comment]
>For normalization should the reward and observation had zero mean and one
variance?

You should not change the mean of your reward (see [nuts and bolts of Deep RL](http://joschu.net/docs/nuts-and-bolts.pdf) linked in the documentation) as it will affect agent’s will to live. The `VecNormalize` wrapper is only scaling the reward.

For continuous actions you should take a look at SAC and TD3, with tuned parameters (cf rl zoo).

**Comment by noyan72** [comment]
The use of  Optuna for optimizing the hyperparameters is useful?

**Comment by araffin** [comment]
>The use of Optuna for optimizing the hyperparameters is useful?

not sure what you mean but it allows automatic hyperparameter tuning which is the recommended way to go.

**Comment by noyan72** [comment]
separate from using every algorithm or library in RL, what tricks can be used for speed up learning? is it depend on complexity of custom environment?
